docs/Conceptual/00 - Introduction.md

```diff
@@ -0,0 +1,188 @@
+
+The way in which this is done is very technology specific, but in essence the environment hosting the tenancy needs to:
+
+1. Identify the intended tenant for the stimulus
+1. Identify the code that is needed to handle the stimulus, and the compute resource to which that stimulus should be directed
+1. Provide a means for that code to find and connect to the services it uses.
+
+There are common patterns that can help a multi-tenanted solution perform these tasks, and abstract as much of this from the code executing in the tenant as possible.
+
+Typically this will include
+1. code at the stimulus/event routing layer (e.g. http routing, message handlers)
+1. factories to be used by the client code to create tenant-specific instances of service clients, configured appropriately
+
+## Solution-layer concerns
+As we dig down in the stack, we finally reach implementation concerns in the multi-tenanted solution itself.
+
+How do you design a SQL Azure database to host multi-tenanted data? What are the isolation characteristics? What about Blob storage? Cosmos DB? 
+
+These solution-level design choices are well explored in the Azure documentation for each individual service. For example
+- [Azure SQL Database]( https://docs.microsoft.com/en-us/azure/azure-sql/database/saas-tenancy-app-design-patterns)
+- [Cosmos DB]( https://docs.microsoft.com/en-us/azure/architecture/guide/multitenant/service/cosmos-db)
+- [Azure Storage]( https://docs.microsoft.com/en-us/azure/architecture/guide/multitenant/service/storage)
+
+When reading that documentation, it is important to consider whether it describes multi-tenant or multi-user scenarios, as we have defined them.
+They sometimes use terms such as *hostile* and *non-hostile* tenants (respectively), to distinguish between these models.
+
+In general, the Azure documentation describes isolation mechanisms for a particular compute or storage service, to support different tenancy and user isolation models, rather than considering a more holistic/solution view of tenancy, bringing together a variety of services into a multi-tenanted solution.
+
+It gives you the nuts-and-bolts (we will talk more about this below), but you will need to understand how each service you consume slots into your overall tenancy model.
+
+For any given technology or service, there are implementation choices which trade-off a number of factors:
+
+1. Security
+    - Does a breach of this service or malicious code execution in one tenant offer a mechanism to hop-over into another tenant?
+1. Noisy neighbours
+    - Does the operation of one tenant have performance or scalability implications for the operation of another tenant? Can one tenant deny or degrade service to others?
+1. Scalability and density
+    - What are the scalability characteristics of the solution as an individual tenant scales up, and/or the number of tenants scales up?
```

Comment:
is sharding a concern here?

Reply:
For particular technologies, yes. I've added a few more "generic" example constraints that allude to that.